In [1]:
import pandas as pd
import pandas as np

policeIncidents = pd.read_csv("Incidents.csv") ## import data set
policeIncidents.head()


PK       CCR  HIERARCHY         INCIDENTTIME  \
0  306530435  23035798          6  2023-03-10T11:30:00   
1  307051774  23038735         10  2023-03-16T09:35:00   
2  307082554  23038816         27  2023-03-16T12:45:00   
3  307106971  23038916         99  2023-03-16T12:00:00   
4  307112015  23038967          6  2023-03-16T17:54:00   

                                   INCIDENTLOCATION CLEAREDFLAG  \
0           600 Block Penn Ave Pittsburgh, PA 15222           N   
1         800 Block Reunion Pl Pittsburgh, PA 15219           N   
2  N Braddock Ave & Alsace Way Pittsburgh, PA 15208           N   
3         1400 Block Locust St Pittsburgh, PA 15219           N   
4        900 Block Liberty Ave Pittsburgh, PA 15222           N   

        INCIDENTNEIGHBORHOOD INCIDENTZONE       INCIDENTHIERARCHYDESC  \
0  Central Business District            2                       THEFT   
1          Bedford Dwellings            2         TERRORISTIC THREATS   
2             Homewood South            5                 HIT AND RUN   
3                      Bluff            2       MISSING PERSONS (18+)   
4              North Oakland            2  RETSIL THEFT (SHOPLIFTING)   

                                            OFFENSES  INCIDENTTRACT  \
0      3921 Theft by Unlawful Taking or Disposition.          201.0   
1    2706 Terroristic Threats. / 2709(a) Harassment.          509.0   
2  3745 Accidents Involving Damage to Unattended ...         1303.0   
3                 9490 Missing Persons (18 and Over)          103.0   
4  3929(a)(1) Retail Theft; takes possession of, ...          403.0   

   COUNCIL_DISTRICT  PUBLIC_WORKS_DIVISION          X          Y  
0               6.0                    6.0 -80.001793  40.442593  
1               NaN                    NaN        NaN        NaN  
2               9.0                    2.0 -79.891415  40.452384  
3               6.0                    3.0 -79.985459  40.436044  
4               6.0                    6.0 -79.997939  40.442945

In [2]:
#new dataframe with only the offense and neighborhood
main_df = policeIncidents[['INCIDENTNEIGHBORHOOD','OFFENSES']].copy()
main_df = main_df.sort_values('INCIDENTNEIGHBORHOOD')

main_df.head(10)

INCIDENTNEIGHBORHOOD                                           OFFENSES
1501     Allegheny Center  3745 Accidents Involving Damage to Unattended ...
233      Allegheny Center                         9496 Dead on Arrival (DOA)
752      Allegheny Center         7615(a)(1) Computer Trespass - Remove Data
62       Allegheny Center  3701 Robbery. / 2702 Aggravated Assault. / 270...
1146     Allegheny Center                               029 Missing Juvenile
922      Allegheny Center  4910 Tampering with or Fabricating Physical Ev...
2215     Allegheny Center                                9489 Found Property
1467     Allegheny Center                             3503 Criminal Trespass
115      Allegheny Center                     9999 9999 - (Misc. Incident): 
1590     Allegheny Center                                9998 Overdose Other

In [4]:
#dict with neighborhood name & number of offenses (occurrences)
    
neighborhood_incidents = main_df.groupby('INCIDENTNEIGHBORHOOD').count().to_dict(orient='dict')['OFFENSES']
for n in neighborhood_incidents:
    print(n, neighborhood_incidents[n])

Allegheny Center 18
Allegheny West 5
Allentown 35
Arlington 20
Arlington Heights 9
Banksville 12
Bedford Dwellings 39
Beechview 45
Beltzhoover 26
Bloomfield 60
Bluff 74
Bon Air 13
Brighton Heights 52
Brookline 67
California-Kirkbride 18
Carrick 84
Central Business District 354
Central Lawrenceville 29
Central North Side 28
Central Oakland 23
Chartiers City 2
Chateau 36
Crafton Heights 27
Crawford-Roberts 65
Duquesne Heights 15
East Allegheny 57
East Carnegie 1
East Hills 32
East Liberty 84
Elliott 37
Esplen 2
Fairywood 10
Fineview 14
Friendship 17
Garfield 35
Glen Hazel 3
Greenfield 21
Hays 3
Hazelwood 47
Highland Park 41
Homewood North 59
Homewood South 61
Homewood West 21
Knoxville 34
Larimer 39
Lincoln Place 9
Lincoln-Lemington-Belmar 61
Lower Lawrenceville 19
Manchester 16
Marshall-Shadeland 71
Middle Hill 36
Morningside 9
Mount Oliver 5
Mount Washington 78
New Homestead 4
North Oakland 38
North Shore 26
Northview Heights 18
Oakwood 10
Outside City 17
Outside County 4
Outside State

# CENTRAL BUSINESS DISTRICT HAS THE MOST CRIMINAL INCIDENTS WITH 354